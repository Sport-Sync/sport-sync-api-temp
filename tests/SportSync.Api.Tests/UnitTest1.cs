using AppAny.HotChocolate.FluentValidation;
using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using SportSync.GraphQL;

namespace SportSync.Api.Tests
{
    public class UnitTest1
    {
        [Fact]
        public async Task SchemaChangeTest()
        {
            var schema = await new ServiceCollection()
                .AddGraphQLServer()
                .AddProjections()
                .AddFiltering()
                .AddAuthorization()
                .AddQueryType<Query>()
                .BuildSchemaAsync();

            schema.ToString().MatchSnapshot();
        }
    }
}