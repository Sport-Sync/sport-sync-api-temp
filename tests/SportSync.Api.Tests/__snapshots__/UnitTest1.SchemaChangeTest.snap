schema {
  query: Query
}

type Query {
  users(where: UserFilterInput): [User] @authorize
}

type User {
  firstName: String
  lastName: String
  email: String
  phone: String
  id: UUID!
}

input StringOperationFilterInput {
  and: [StringOperationFilterInput!]
  or: [StringOperationFilterInput!]
  eq: String
  neq: String
  contains: String
  ncontains: String
  in: [String]
  nin: [String]
  startsWith: String
  nstartsWith: String
  endsWith: String
  nendsWith: String
}

input UserFilterInput {
  and: [UserFilterInput!]
  or: [UserFilterInput!]
  firstName: StringOperationFilterInput
  lastName: StringOperationFilterInput
  email: StringOperationFilterInput
  phone: StringOperationFilterInput
  id: UuidOperationFilterInput
}

input UuidOperationFilterInput {
  eq: UUID
  neq: UUID
  in: [UUID]
  nin: [UUID]
  gt: UUID
  ngt: UUID
  gte: UUID
  ngte: UUID
  lt: UUID
  nlt: UUID
  lte: UUID
  nlte: UUID
}

enum ApplyPolicy {
  BEFORE_RESOLVER
  AFTER_RESOLVER
  VALIDATION
}

directive @authorize("The name of the authorization policy that determines access to the annotated resource." policy: String "Roles that are allowed to access the annotated resource." roles: [String!] "Defines when when the authorize directive shall be applied.By default the authorize directives are applied during the validation phase." apply: ApplyPolicy! = BEFORE_RESOLVER) repeatable on OBJECT | FIELD_DEFINITION

scalar UUID @specifiedBy(url: "https:\/\/tools.ietf.org\/html\/rfc4122")
